### PR TITLE
Kubectl port-forward doesn't warn about used port? #664

### LINKED ISSUE #664: Kubectl port-forward doesn't warn about used port?
eg. I have a service on local machine run under port `5555`. 
Why then the command: `kubectl port-forward {{pod_name}} 5555:{{some_port}}` is allowed?

Isn't it supposed to warn the user about: `port 5555 is already used`?

### PR DESCRIPTION
**Pull request by pswica**
**What type of PR is this?**

> /kind bug

**What this PR does / why we need it**:
Fixes kubernetes/kubectl#664

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Fixes a bug where kubectl port-forward allowed forwarding ports that were already in use. kubectl port-forward also now requires sudo to use privileged ports (80, 443, ...)
```

### CODE REVIEW

File: pkg/kubectl/cmd/portforward/portforward.go [hunk 1]
```diff
@@ -210,6 +212,11 @@ func convertPodNamedPortToNumber(ports []string, pod corev1.Pod) ([]string, erro
 
 			containerPortStr = strconv.Itoa(int(containerPort))
 		}
+		ln, err := net.Listen("tcp", ":"+localPort)
+		if err != nil {
+			return nil, err
```
Comment (draveness) [inline pkg/kubectl/cmd/portforward/portforward.go]: I'm curious about the error message here when the port is in use. Maybe we could wrap the error with additional information to clarify what happens here.

/priority backlog
/ok-to-test

Comment (pswica) [reply pkg/kubectl/cmd/portforward/portforward.go]: Thanks for reviewing this! I was thinking the same thing, but the error message ends up being very helpful:
`error: listen tcp :3306: bind: address already in use`

**Comment by pswica** [comment]
This is an interesting test failure, since I check if the system can listen on a port. It failed because Linux systems normally require `sudo` privileges to listen on port 80--if I run `sudo kubectl port-forward <pod> 80` it works, as expected. This makes me think that, for security reasons, the test should be checking another port. Perhaps I will try another PR for this.

**Comment by pswica** [comment]
Following some reviews https://github.com/kubernetes/kubernetes/pull/78809, I decided to address https://github.com/kubernetes/kubectl/issues/666 over here. That is, I decided to change the test on privileged ports on this PR as well

**Comment by neolit123** [comment]
to make the PR close the related issue this:
```
Fixes # kubernetes/kubectl#664
```
has to be changed to:
```
Fixes kubernetes/kubectl#664
```

i understand the motivation of the PR, but i would defer to the kubectl maintainers for approval.
thanks.

/unassign

**Comment by pswica** [comment]
@kubernetes/sig-cli-pr-reviews PTAL

**Comment by pswica** [comment]
@soltysh PTAL

**Pull request closed without merging**